diff --git a/web/pgadmin/static/js/helpers/Notifier.jsx b/web/pgadmin/static/js/helpers/Notifier.jsx
index 8338dfbdd..ad1591b8f 100644
--- a/web/pgadmin/static/js/helpers/Notifier.jsx
+++ b/web/pgadmin/static/js/helpers/Notifier.jsx
@@ -190,7 +190,9 @@ export function NotifierProvider({ pgAdmin, pgWindow, getInstance, children, onR
   useEffect(()=>{
     // if open in an iframe then use top pgAdmin
     if(window.self != window.top) {
-      pgAdmin.Browser.notifier = new Notifier(modal, pgWindow.pgAdmin.Browser.notifier.snackbar);
+      if (pgWindow) {
+        pgAdmin.Browser.notifier = new Notifier(modal, pgWindow.pgAdmin.Browser.notifier.snackbar);
+      }
       onReady?.();
       getInstance?.(pgAdmin.Browser.notifier);
     }
